azure-pipelines/end-to-end-tests-dir/fetch.ps1

```diff
@@ -6,90 +6,107 @@ if (-not $IsMacOS -and -not $IsLinux) {
-        <archiveName>cmake-3.20.4.txz</archiveName>
-    </tool>
-</tools>
-"@ | % { $_ -replace "`r","" } | Out-File -enc ascii $(Join-Path $Scripts "vcpkgTools.xml")
+    $7zip_version = "19.00"
+    $ninja_version = "1.10.2"
+
+    @'
+{
+    "schema-version": 1,
+    "tools": [{
+        "name": "7zip",
+        "os": "windows",
+        "version": "19.00",
+        "executable": "Files\\7-Zip\\7z.exe",
+        "url": "https://www.7-zip.org/a/7z1900-x64.msi",
+        "sha512": "7837a8677a01eed9c3309923f7084bc864063ba214ee169882c5b04a7a8b198ed052c15e981860d9d7952c98f459a4fab87a72fd78e7d0303004dcb86f4324c8",
+        "archive": "7z1900-x64.msi"
+    },
+    {
+        "name": "ninja-testing",
+        "os": "windows",
+        "version": "1.10.2",
+        "executable": "ninja.exe",
+        "url": "https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-win.zip",
+        "sha512": "6004140d92e86afbb17b49c49037ccd0786ce238f340f7d0e62b4b0c29ed0d6ad0bab11feda2094ae849c387d70d63504393714ed0a1f4d3a1f155af7a4f1ba3",
+        "archive": "ninja-win-1.10.2.zip"
+    },
+    {
+        "name": "ninja",
+        "os": "windows",
+        "version": "1.10.2",
+        "executable": "ninja.exe",
+        "url": "https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-win.zip",
+        "sha512": "6004140d92e86afbb17b49c49037ccd0786ce238f340f7d0e62b4b0c29ed0d6ad0bab11feda2094ae849c387d70d63504393714ed0a1f4d3a1f155af7a4f1ba3",
+        "archive": "ninja-win-1.10.2.zip"
+    },
+    {
+        "name": "cmake",
+        "os": "windows",
+        "version": "3.22.2",
+        "executable": "cmake-3.22.2-windows-i386\\bin\\cmake.exe",
+        "url": "https://github.com/Kitware/CMake/releases/download/v3.22.2/cmake-3.22.2-windows-i386.zip",
+        "sha512": "969d3d58d56d8fa3cc3acae2b949bf58abab945f70ae292ff20c9060d845dfc094c613c367a924abff47f307cc33af1467cdb9b75bb857868e38b2c7cdc72f79",
+        "archive": "cmake-3.22.2-windows-i386.zip"
+    },
+    {
+        "name": "cmake",
+        "os": "osx",
+        "version": "3.22.2",
+        "executable": "cmake-3.22.2-macos-universal/CMake.app/Contents/bin/cmake",
+        "url": "https://github.com/Kitware/CMake/releases/download/v3.22.2/cmake-3.22.2-macos-universal.tar.gz",
+        "sha512": "08104f608ecb9a5cfef38e79f0957d21e425616c0677781445492f82cbfec805113e3b5eb4bc737b707bb26a00678e7bd55e17555a5611c08b0b9b44ac5136ac",
+        "archive": "cmake-3.22.2-macos-universal.tar.gz"
+    },
+    {
+        "name": "cmake",
+        "os": "linux",
+        "version": "3.22.2",
+        "executable": "cmake-3.22.2-linux-x86_64/bin/cmake",
+        "url": "https://github.com/Kitware/CMake/releases/download/v3.22.2/cmake-3.22.2-linux-x86_64.tar.gz",
+        "sha512": "579e08b086f6903ef063697fca1dc2692f68a7341dd35998990b772b4221cdb5b1deecfa73bad9d46817ef09e58882b2adff9d64f959c01002c11448a878746b",
+        "archive": "cmake-3.22.2linux-x86_64.tar.gz"
```

Comment:
Since this entry has no architecture entry this version would be used on arm64 devices. Maybe the arch field should be mandatory.